### PR TITLE
feat(genqa): update citations

### PR DESCRIPTION
**Pull request by jelmedini**
[SVCC-3999](https://coveord.atlassian.net/browse/SVCC-3999)

Improve CRGA citations:

- Remove numbered citations.

Before:

Screenshot 2024-07-31 at 2 34 39 PM

After:

Screenshot 2024-07-31 at 2 54 26 PM

[SVCC-3999]: https://coveord.atlassian.net/browse/SVCC-3999?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: packages/atomic/src/components/common/generated-answer/generated-answer-common.tsx [hunk 1]
```diff
@@ -195,7 +195,17 @@ export class GeneratedAnswerCommon {
...
+      (acc: GeneratedAnswerCitation[], citation) => {
+        if (!acc.some((item) => item?.clickUri === citation.clickUri)) {
+          acc.push(citation);
+        }
+        return acc;
+      },
+      []
+    );
+
+    return uniqueCitations?.map(
```
Comment (louis-bompart) [inline packages/atomic/src/components/common/generated-answer/generated-answer-common.tsx:208]: Double array iterations.
An Array.flatMap should do the job with a single array iteration

File: packages/atomic/src/components/common/generated-answer/generated-answer-common.tsx [hunk 2]
```diff
@@ -195,26 +195,34 @@ export class GeneratedAnswerCommon {
     const {citations} = getGeneratedAnswerState() ?? {};
     const {logCitationHover} = getGeneratedAnswer() ?? {};
 
-    return citations?.map(
+    const uniqueUris = new Set<string>();
+
+    return citations?.flatMap(
```
Comment (erocheleau) [inline packages/atomic/src/components/common/generated-answer/generated-answer-common.tsx:200]: Why isn't this handled in Headless?

Comment (jelmedini) [reply packages/atomic/src/components/common/generated-answer/generated-answer-common.tsx:200]: @erocheleau Very good point! I honestly didnt think about it, as we are changing also a little bit the UI in atomic I did it here.
I could still change to headless but we still need to change the numbered citations in the UI for both!

**Review by dmgauthier — APPROVED** [review]

File: packages/atomic/src/components/common/generated-answer/generated-answer-common.tsx (hunk 2 above)
Comment (louis-bompart) [reply packages/atomic/src/components/common/generated-answer/generated-answer-common.tsx]: This branch should probably be based on #4250 now, I reckon, and the changes on Atomic to take into consideration those made in Headless.

**Review by lvu285 — APPROVED** [review]

**Pull request merged**